.github/workflows/cov.yml

```diff
@@ -0,0 +1,105 @@
+name: Daily Coverage
+
+on:
+    schedule:
+      - cron: '0 6 * * *' # run at 6 AM UTC
+    workflow_dispatch:
+
+jobs:
+  pre-commit:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v3
+        with:
+          fetch-depth: 2
+      - uses: actions/setup-python@v3
+      - name: Install dependencies
+        run: |
+          python -m pip install pre-commit
+          python -m pip freeze --local
+      - uses: actions/cache@v3
+        with:
+          path: ~/.cache/pre-commit
+          key: pre-commit|${{ env.pythonLocation }}|${{ hashFiles('.pre-commit-config.yaml') }}
+      - name: Run pre-commit checks
+        run: pre-commit run --show-diff-on-failure --color=always --from-ref HEAD^ --to-ref HEAD
```

Comment:
Don't we run `pre-commit` checks anyway on each PR? I think the `pre-commit` job is redundant here and it's safe to remove this from the `cov.yml`